source/Planet.cpp

```diff
@@ -188,6 +188,8 @@ void Planet::Load(const DataNode &node, const Set<Sale<Ship>> &ships, const Set<
 		else
 			attributes.erase(AUTO_ATTRIBUTES[i]);
 	}
+
+	inhabited = (HasSpaceport() || requiredReputation || defenseFleet) && !attributes.count("uninhabited");
```

Comment:
OK.
A planet's inhabited status is only changed in the `Planet::Load()`, oddly enough.